apps/mull-ui/src/app/pages/create-event/location-autocomplete/location-autocomplete-textbox.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useState, ChangeEvent, useEffect, useMemo } from 'react';
+import TextField from '@material-ui/core/TextField';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import { gql, useLazyQuery } from '@apollo/client';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import './location-autocomplete-textbox.scss';
+import { faMapMarkerAlt, faLocationArrow } from '@fortawesome/free-solid-svg-icons';
+import { debounce } from 'lodash';
+
+export const AUTOCOMPLETED_LOCATIONS = gql`
+  query Query($userInput: String!) {
+    getAutocompletedLocations(userInput: $userInput)
+  }
+`;
+
+export interface LocationAutocompleteTextboxProps {
+  handleClose: (value: string) => void;
+  input: string;
+}
+
+export default function LocationAutocompleteTextbox({ handleClose, input }) {
+  const [open, setOpen] = useState(false);
+  const [options, setOptions] = useState<string[]>([]);
+  const [getAutocompletedLocations, { loading, data }] = useLazyQuery(AUTOCOMPLETED_LOCATIONS);
+
+  const debounceGetLocation = useMemo(
```

Comment:
I made this function. Here's an explanation:

- debounce
A debounced function will only fire once it has not been called after a set delay. In this example, this function will be called on every keystroke in the autocomplete textbox, but the function will not actually run. Once the user stops typing, it will call the function with the text that was inputted.

- useMemo
Since this is a React component, it will re-render on any change to the state. Therefore, the debounced function will be recreated on every re-render. Using the useMemo hook, we are able to memoize the function so that it is not reset every time.

This essentially cuts down the number of requests being sent as we are only sending a request to the geolocation API once the user has finished typing.